unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/NavMap/NavmapView.cs

```diff
@@ -7,32 +7,120 @@ namespace DCL
-        bool enabledInProduction = false;
+        bool enabledInProduction = true;
 
         void Start()
         {
+            mapMetadata = MinimapMetadata.GetMetadata();
+
+            worldmapOffset = new Vector3(LEFT_BORDER_PARCELS + WORLDMAP_WIDTH_IN_PARCELS / 2, BOTTOM_BORDER_PARCELS + WORLDMAP_WIDTH_IN_PARCELS / 2, 0);
+            parcelSizeInMap = (MapUtils.PARCEL_SIZE / 2);
+
+            parcelHighlightImage = MapRenderer.i.parcelHighlightImage;
+            highlightedParcelText = parcelHighlightImage.GetComponentInChildren<TextMeshProUGUI>(true);
+            parcelHighlightImage.rectTransform.localScale = new Vector3(parcelHightlightScale, parcelHightlightScale, 1f);
+
             closeButton.onClick.AddListener(() => { ToggleNavMap(); });
-            scrollRect.onValueChanged.AddListener((x) => { if (scrollRect.gameObject.activeSelf) MapRenderer.i.atlas.UpdateCulling(); });
+            scrollRect.onValueChanged.AddListener((x) => { if (isOpen) MapRenderer.i.atlas.UpdateCulling(); });
 
             toggleNavMapDelegate = (x) => { ToggleNavMap(); };
             toggleNavMapAction.OnTriggered += toggleNavMapDelegate;
 
             MinimapHUDView.OnUpdateData += UpdateCurrentSceneData;
         }
 
+        void Update()
+        {
+            if (!isOpen) return;
+
+            // Get the world-space corners of the map
+            (MapRenderer.i.atlas.chunksParent.transform as RectTransform).GetWorldCorners(navmapWorldspaceCorners);
+
+            // Offset world coordinates origin position in map with border-parcels and worldmap amount of parcels (horizontally/vertically) / 2
+            // (since the "border-parcels" outside the world are not the same amount on the 4 sides of the worldmap we can't just use the center of the rect)
+            worldCoordsOriginInMap = navmapWorldspaceCorners[0] + worldmapOffset * parcelSizeInMap;
+
+            UpdateMouseMapCoords();
+
+            UpdateParcelHighlight();
+        }
+
+        void UpdateMouseMapCoords()
+        {
+            mouseMapCoords = Input.mousePosition - worldCoordsOriginInMap;
+            mouseMapCoords = mouseMapCoords / parcelSizeInMap;
+
+            mouseMapCoords.x = (int)Mathf.Floor(mouseMapCoords.x);
+            mouseMapCoords.y = (int)Mathf.Floor(mouseMapCoords.y);
+        }
+
+        void UpdateParcelHighlight()
+        {
+            if (!CoordinatesAreInsideTheworld((int)mouseMapCoords.x, (int)mouseMapCoords.y))
+            {
+                if (parcelHighlightImage.gameObject.activeSelf)
+                    parcelHighlightImage.gameObject.SetActive(false);
+
+                return;
+            }
+
+            if (!parcelHighlightImage.gameObject.activeSelf)
+                parcelHighlightImage.gameObject.SetActive(true);
+
+            parcelHighlightImage.transform.position = worldCoordsOriginInMap + mouseMapCoords * parcelSizeInMap + new Vector3(parcelSizeInMap, parcelSizeInMap, 0f) / 2;
```

Comment:
All the logic related to positioning the image as intended should be in `MapRenderer`, as that class should take all the rendering responsibilities.

Reply:
Done, check it out